src/AppBundle/Controller/Admin/BlogController.php

```diff
@@ -147,6 +148,7 @@ public function editAction(Post $post, Request $request)
         if ($form->isSubmitted() && $form->isValid()) {
             $post->setSlug($this->get('slugger')->slugify($post->getTitle()));
             $entityManager->flush();
+            $this->getDoctrine()->getRepository(Tag::class)->cleanUnusedTags();
```

Comment:
you could make this code much more efficient, by getting the old list of tags of the Post in an array before ``handleRequest``, then comparing it to the list of new tags here. The difference will give you the list of tags removed from the post, which are the only ones you need to check for being unused